javatests/arcs/core/analysis/RecipeGraphTest.kt

```diff
@@ -205,4 +207,63 @@ class RecipeGraphTest {
             testAllConnections(it)
         }
     }
+
+    @Test
+    fun particleNodes() {
+        with (TestRecipe()) {
+            val graph = RecipeGraph(recipe)
+            assertThat(graph.particleNodes.map { it.particle }).containsExactly(
+                readerParticle,
+                writerParticle
+            )
+        }
+    }
+
+    @Test
+    fun handleNodes() {
+        with (TestRecipe()) {
+            val graph = RecipeGraph(recipe)
+            assertThat(graph.handleNodes.map { it.handle }).containsExactly(
+                thingHandle,
+                someHandle,
+                joinedHandle
+            )
+        }
+    }
+
+    @Test
+    fun particleNode_claimsAreMutable() {
```

Comment:
It's a weird test to have I think - as soon as you start implementing the algorithm, it would not compile if checks/claims were not mutable, so you won't need to additionally test it here. WDYT?

Reply:
Removed